File: mambo-protocol/src/main/scala/org/mambo/protocol/messages/game/context/roleplay/party/PartyLeaderUpdateMessage.scala





















// Generated on 05/03/2013 10:34:47
// Scala version by Blackrush
package org.mambo.protocol.messages

import org.mambo.core.io._
import org.mambo.protocol.{Message => BaseMessage, _}
import org.mambo.protocol.types._
import org.mambo.protocol.enums._

trait PartyLeaderUpdateMessage extends AbstractPartyEventMessage {
  override val messageId = 5578

  protected trait Message extends AbstractPartyEventMessage.Message with Serializable {
    override val messageId = PartyLeaderUpdateMessage.messageId

    val partyLeaderId: Int

    override def serialize(buf: Buffer) {
      super.serialize(buf)

      buf.writeInt(partyLeaderId)
    }

  }

  protected def create(partyId: Int, partyLeaderId: Int): Target = throw new IllegalAccessException

  override def deserialize(buf: Buffer) = {
    val partyId = buf.readInt()
    val partyLeaderId = buf.readInt()

    create(partyId, partyLeaderId)
  }

}

object PartyLeaderUpdateMessage extends PartyLeaderUpdateMessage {
  case class MessageImpl(partyId: Int, partyLeaderId: Int) extends Message
  type Target = MessageImpl

  override def create(partyId: Int, partyLeaderId: Int) = MessageImpl(partyId, partyLeaderId)

  def apply(partyId: Int, partyLeaderId: Int) = create(partyId, partyLeaderId)
  def unapply(msg: MessageImpl) = MessageImpl.unapply(msg)
}

File: mambo-protocol/src/main/java/org/mambo/protocol/messages/PartyLeaderUpdateMessage.java


// Generated on 05/08/2013 19:37:49
package org.mambo.protocol.messages;

import java.util.*;
import org.mambo.protocol.types.*;
import org.mambo.protocol.enums.*;
import org.mambo.protocol.*;
import org.mambo.core.io.*;

public class PartyLeaderUpdateMessage extends AbstractPartyEventMessage {
    public static final int MESSAGE_ID = 5578;
    
    @Override
    public int getNetworkMessageId() {
        return MESSAGE_ID;
    }
    
    public int partyLeaderId;
    
    public PartyLeaderUpdateMessage() { }
    
    public PartyLeaderUpdateMessage(int partyId, int partyLeaderId) {
        super(partyId);
        this.partyLeaderId = partyLeaderId;
    }
    
    @Override
    public void serialize(Buffer buf) {
        super.serialize(buf);
        buf.writeInt(partyLeaderId);
    }
    
    @Override
    public void deserialize(Buffer buf) {
        super.deserialize(buf);
        partyLeaderId = buf.readInt();
        if (partyLeaderId < 0)
            throw new RuntimeException("Forbidden value on partyLeaderId = " + partyLeaderId + ", it doesn't respect the following condition : partyLeaderId < 0");
    }
    
}
